crates/subspace-networking/src/node_runner.rs

```diff
@@ -173,14 +227,7 @@ impl NodeRunner {
                 endpoint,
                 ..
             } => {
-                let shared = match self.shared_weak.upgrade() {
-                    Some(shared) => shared,
-                    None => {
-                        return;
-                    }
-                };
                 debug!("Connection established with peer {peer_id} [{num_established} from peer]");
-                shared.connected_peers_count.fetch_add(1, Ordering::SeqCst);
```

Comment:
Why did you remove it? It was one atomic read away in `Node`, now you need to make a request over the channel to know how many connected peers there is. We actually want more of these variables, not less IMO.